### PR TITLE
feat(dagger): Support workflow-name on dagger module attestations

### LINKED ISSUE #846: Support new attestation modality in Dagger Module
Our dagger module doesn't support the modality where the workflow name is provided on `init`.

### PR DESCRIPTION
**Pull request by javirln**
This patch allows to pass `--workflow-name` to attestations on the Dagger module when using API tokens.

Example calling attestation status:
```
$ dagger call -m . init --token env:CHAINLOOP_TOKEN --workflow-name dagger-wf status
┌───────────────────┬──────────────────────────────────────┐
│ Initialized At    │ 04 Jun 24 18:16 UTC                  │
├───────────────────┼──────────────────────────────────────┤
│ Attestation ID    │ 045bf795-b44c-4a7f-828f-d1dea1af40f7 │
│ Name              │ dagger-wf                            │
│ Team              │ founding                             │
│ Project           │ core                                 │
│ Contract Revision │ 1                                    │
└───────────────────┴──────────────────────────────────────┘
```
Adding a file-evidence:
```bash
$ dagger call -m . init --token env:CHAINLOOP_TOKEN --workflow-name dagger-wf add-file-evidence --name artifact --path ./CONTRIBUTING.md push --key file:./cosign.key --passphrase env:COSIGN_PASSWORD
┌───────────────────┬──────────────────────────────────────┐
│ Initialized At    │ 05 Jun 24 06:07 UTC                  │
├───────────────────┼──────────────────────────────────────┤
│ Attestation ID    │ 020286de-fc11-4d44-a9bc-97b6ecc16153 │
│ Name              │ dagger-wf                            │
│ Team              │ founding                             │
│ Project           │ core                                 │
│ Contract Revision │ 2                                    │
└───────────────────┴──────────────────────────────────────┘
┌─────────────────────────────────────────────────────────────────────────────────────┐
│ Materials                                                                           │
├───────────┬─────────────────────────────────────────────────────────────────────────┤
│ Name      │ artifact                                                                │
│ Type      │ ARTIFACT                                                                │
│ Set       │ Yes                                                                     │
│ Required  │ Yes                                                                     │
│ Is output │ Yes                                                                     │
│ Value     │ CONTRIBUTING.md                                                         │
│ Digest    │ sha256:8ad74d448581170e408bc4f5cb797548b7d5ea9e1ddff81dc0a7cb8672daeb6a │
└───────────┴─────────────────────────────────────────────────────────────────────────┘
┌───────────────────────────────────┐
│ Runner context                    │
├─────────────────────────┬─────────┤
│ CHAINLOOP_DAGGER_CLIENT │ v0.90.1 │
└─────────────────────────┴─────────┘
Attestation Digest: sha256:fe2a03885b7c108df54985dce3f80c71645f5f2fac5dbf02192367877c94a1eb
```
Example adding raw-evidence:
```bash
$ dagger call -m . init --token env:CHAINLOOP_TOKEN --workflow-name dagger-wf add-raw-evidence --name string --value "my-value" push --key file:./cosign.key --passphrase env:COSIGN_PASSWORD
┌───────────────────┬──────────────────────────────────────┐
│ Initialized At    │ 05 Jun 24 06:09 UTC                  │
├───────────────────┼──────────────────────────────────────┤
│ Attestation ID    │ 13758623-7bb1-40b3-ab89-e3a3d5973061 │
│ Name              │ dagger-wf                            │
│ Team              │ founding                             │
│ Project           │ core                                 │
│ Contract Revision │ 4                                    │
└───────────────────┴──────────────────────────────────────┘
┌──────────────────────┐
│ Materials            │
├───────────┬──────────┤
│ Name      │ string   │
│ Type      │ STRING   │
│ Set       │ Yes      │
│ Required  │ Yes      │
│ Is output │ Yes      │
│ Value     │ my-value │
└───────────┴──────────┘
┌───────────────────────────────────┐
│ Runner context                    │
├─────────────────────────┬─────────┤
│ CHAINLOOP_DAGGER_CLIENT │ v0.90.1 │
└─────────────────────────┴─────────┘
Attestation Digest: sha256:7c12e7fcc872ef886dd17fe106946417a227c45c9a28bec55d55562e9915ff6c
```

Closes #846

### CODE REVIEW

**Comment by javirln** [comment]
A new release of Chainloop containing the change https://github.com/chainloop-dev/chainloop/pull/874 needs to be cut in order to test and merge it.

cc @jiparis @migmartri

File: extras/dagger/main.go [hunk 1]
```diff
@@ -49,6 +49,10 @@ func (m *Chainloop) Init(
 	// Path to the source repository to be attested
 	// +optional
 	repository *Directory,
+	// Workflow name to be used for the attestation, default empty
+	// This option is needed when using an API Token
+	// +optional
+	workflowName string,
```
Comment (migmartri) [inline extras/dagger/main.go:55]: I am wondering if we should make this required and basically be the only mode supported for dagger.

File: extras/dagger/README.md [hunk 2]
```diff
@@ -73,6 +73,18 @@ dagger call -m github.com/chainloop-dev/chainloop \
   --contract-revision 1 # optional flag to specify the revision of the Workflow Contract (default `latest`)
 ```
 
+If the token passed is an API Token, the name of the workflow needs to be pass as well.
```
Comment (migmartri) [inline extras/dagger/README.md:76]: > API Token and not a ROBOT-ACCOUNT

**Review by migmartri — COMMENTED** [review]
Thanks

File: extras/dagger/main.go (hunk 1 above)
Comment (javirln) [reply extras/dagger/main.go:55]: Personally I would mark it as required once we have all pipelines at least migrated to the API tokens

Comment (migmartri) [reply extras/dagger/main.go:55]: Why is that? Are you concerned of the feature readiness itself to do not want to confuse people with out to date documentation?

Comment (javirln) [reply extras/dagger/main.go:55]: It’s about documentation, I want the desync with it to last the bear minimum if possible

**Review by migmartri — COMMENTED** [review]
🚀

**Pull request merged**